libdnf/base/goal.cpp

```diff
@@ -905,6 +936,62 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
     }
 }
 
+GoalProblem Goal::Impl::add_group_install_to_goal(
+    base::Transaction & transaction,
+    const std::string & spec,
+    const transaction::TransactionItemReason reason,
+    GroupJobSettings & settings) {
+    auto & cfg_main = base->get_config();
+    bool strict = settings.resolve_strict(cfg_main);
+    comps::GroupQuery group_query(base);
+    comps::GroupQuery group_query_names(group_query);
+    group_query_names.filter_name(spec, sack::QueryCmp::IGLOB);
+    group_query.filter_groupid(spec, sack::QueryCmp::IGLOB);
+    group_query |= group_query_names;
+    if (group_query.empty()) {
+        auto problem = transaction.p_impl->report_not_found(GoalAction::INSTALL, spec, GoalJobSettings(), strict);
```

Comment:
It will be quite difficult to report group problem in current goal log. I believe that it is a right place but the place is not optimized to accept GroupJobSettings. May be we can discuss it.